src/components/application_manager/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -103,8 +104,9 @@ void AddSubMenuRequest::Run() {
   const mobile_apis::ImageType::eType type =
       static_cast<mobile_apis::ImageType::eType>(image_type);
   const bool is_dynamic_image = mobile_apis::ImageType::DYNAMIC == type;
-
-  if (is_key_icon_exist && is_icon_path_valid && is_dynamic_image) {
+  if (!is_key_icon_exist) {
```

Comment:
@pold500 
I recommend you just not to add the icon parameter part of the message if the icon is not valid.